### DISCUSSION TITLE
Where to put cspell?

### DISCUSSION THREAD

**Discussion by cyntheticfox** [w0rp/ale#3970] [Q&A - Ask for help with problems] [answered]
So, I went to go look at writing a linter implementation for cspell for #3828, and got as far as writing a simple handler and tests, when I remembered some of the tools like `write-good` and `alex` are present in multiple language directories, since they support general text formats.

Where would cspell go? It supports [a lot](https://github.com/streetsidesoftware/cspell/blob/main/packages/cspell-bundled-dicts/package.json#L45) of languages by default, and more importantly, [allows even more](https://github.com/streetsidesoftware/cspell-dicts) to be added with `cspell link`.

Would it be possible to just drop a `cspell.vim` in `/ale_linters/`? Manually adding it for every language seems...impractical, especially since you can add support for new ones dynamically.

**Comment by hsanson** [comment, marked as answer]
If you look at writegood implementation you can see all methods are defined inside `autoload/ale/handlers/writegood.vim` and then there are specific files for each language:

 - ale_linters/markdown/writegood.vim 
 - ale_linters/asciidoc/writegood.vim
 - ale_linters/texinfo/writegood.vim
 - ale_linters/nroff/writegood.vim
 - ale_linters/xhtml/writegood.vim
 - ale_linters/html/writegood.vim
 - etc...

And each of these only call one method to define the linter for the respective language:

```
call ale#handlers#writegood#DefineLinter('po')
```
